utils/trainer.py

```diff
@@ -0,0 +1,169 @@
+import json
+import time
+from typing import List
+import logging
+from pathlib import Path
+
+from tqdm import tqdm
+import torch
+import torch.distributed as dist
+
+from utils.utils import compute_loss
+from utils.torch_utils import SequentialDistributedSampler
+import utils.torch_utils as torch_utils
+from utils.utils import non_max_suppression, scale_coords, xywh2xyxy, xyxy2xywh, clip_coords, plot_images, output_to_target
+
+
+class Trainer:
```

Comment:
Could this be renamed to something else? Like Inferencer instead? I think it would be good to leave the "Trainer" class name for optimizing the train file.

Reply:
I'm currently trying to see refactor the `train` file, but in a different way. I break down the `def train` into smaller functions belonging to a `Trainer` class. I'm not sure if this is a good idea, but I will share about it soon.

Reply:
I assume the trainer class is used both for train and evaluation, as the same pattern in pytorch_lightning. The future train code can be directly added to trainer class.